lib/LedgerSMB/Upgrade_Tests.pm

```diff
@@ -173,19 +188,50 @@ has display_name => (is => 'ro', isa => 'Str', required => 1);
 
 Human readable instructions for test, localized.
 
-=back
-
 =cut
 
 has instructions => (is => 'ro', isa => 'Str', required => 1);
 
+=item buttons
+
+Enabled buttons
+
+=cut
+
+subtype 'button'
+    => as 'Str'
+    => where { $_ =~ /Save and Retry|Cancel|Force/ }
+    => message { "Invalid button '$_'" };
+
+has buttons => (is => 'ro', isa => 'ArrayRef[button]',
+                default => sub {['Save and Retry', 'Cancel']},
+                required => 0);
+
+=item tooltips
+
+Tooltip for each button
+
+=cut
+
+has tooltips => (is => 'ro',
+    isa => 'HashRef[Str]',
+    default => sub {
+        return {
+            'Save and Retry' => marktext('Save the fixes provided and attempt to continue migration'),
+            'Cancel' => marktext('Cancel the migration')
+    }},
+    required => 0);
+
+=back
+
+=head1 Methods
```

Comment:
For next time: `head1`s are always in full capitals in our code (and when they're not, that's a bug...)